azurerm/internal/services/containers/kubernetes_cluster_pod_identity_resource_test.go

```diff
@@ -0,0 +1,239 @@
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (t KubernetesClusterPodIdentityResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.ClusterID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Containers.KubernetesClustersClient.Get(ctx, id.ResourceGroup, id.ManagedClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("reading Kubernetes Cluster (%s): %+v", id.String(), err)
+	}
+
+	if resp.ManagedClusterProperties == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile == nil ||
+		resp.ManagedClusterProperties.PodIdentityProfile.Enabled == nil ||
+		!*resp.ManagedClusterProperties.PodIdentityProfile.Enabled {
+		return utils.Bool(false), nil
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r KubernetesClusterPodIdentityResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  depends_on = [
+    azurerm_role_assignment.test
+  ]
+}
+`, r.template(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "import" {
+  cluster_id = azurerm_kubernetes_cluster_pod_identity.test.cluster_id
+
+  dynamic "pod_identity" {
+    for_each = azurerm_kubernetes_cluster_pod_identity.test.pod_identity
+    content {
+      name        = pod_identity.value.name
+      namespace   = pod_identity.value.namespace
+      identity_id = pod_identity.value.identity_id
+    }
+  }
+}
+`, r.basic(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  exception {
+    name      = "exception1"
+    namespace = "exception-ns1"
+    pod_labels = {
+      "env" : "test"
+    }
+  }
+
+  depends_on = [
+    azurerm_role_assignment.test
+  ]
+}
+`, r.template(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  linux_profile {
```

Comment:
Is this required for the test?

Reply:
yes

Reply:
Persumably this is missing in the document?